### PR TITLE
Created Homepage controller and html page

### LINKED ISSUE #2: Homepage view
# REQUIREMENTS
Layout of the Homepage view (Check-in and check-out time stamp)

Create controller for this view

# DESIGN
The layout of this view should look like this mockup
image

### PR DESCRIPTION
**Pull request by corps01**
rails generate controller Home index

We created a hompage view using rails generate controller, the homepage includes an animated clock created with sass and js
Sass and JS allow us to print a realtime digital clock and annimate an analog clock.

Co-authored-by: dany-herz42 <danyherzdev@gmail.com>

This route to testing: 
/home/index

### CODE REVIEW

File: checkClock/app/assets/stylesheets/home.scss [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+          }
+        }
+      }
+    }
+  }
+
+  .navigation {
+    background-color: $color-secondary;
+    border-radius: 15px;
+    font-family: "Poppins", sans-serif;
```
Comment (JuanequeX) [inline checkClock/app/assets/stylesheets/home.scss]: You can also create specific variables for your fonts, same like yours colors.

File: checkClock/app/assets/stylesheets/home.scss [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+    padding: 10px 20px;
+    cursor: pointer;
+    a {
+      color: white;
```
Comment (JuanequeX) [inline checkClock/app/assets/stylesheets/home.scss]: This color can be a variable

File: checkClock/app/views/home/index.html.erb [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+          <button class="button-success">Check In</button>
+          <button class="button-danger">Check Out</button>
+        </div>
+      </div>
+    </div>
+  </main>
+</body>
+<%= javascript_pack_tag 'homepage' %>
+<%= javascript_pack_tag 'analogic' %>
+</html>
```
Comment (JuanequeX) [inline checkClock/app/views/home/index.html.erb]: Add a blank line here

File: checkClock/app/javascript/packs/homepage.js [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+  const hour = document.querySelector('#hour');
+  hour.textContent = `${getTimeStr(currentDate.getHours())}:${getTimeStr(currentDate.getMinutes())}:${getTimeStr(currentDate.getSeconds())}`
+}
+
+function getTimeStr(value){
+  value < 10 ? value = `0${value}` : null
+  return value
+}
+
+setInterval(printHour,1000)
```
Comment (JuanequeX) [inline checkClock/app/javascript/packs/homepage.js]: Add a blank line here

File: checkClock/app/javascript/packs/analogic.js [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+    }
+    let degHours = (360 * hour / 12)- 180
+    let degMiutes = 180 + minutes * 6
+    const manecillaHour = document.querySelector(".hours-logo")
+    const manecillaMinutos = document.querySelector(".minutes-logo")
+    manecillaHour.style.transform = `translate(-50%) rotate(${degHours}deg)`
+    manecillaMinutos.style.transform = `translate(-50%) rotate(${degMiutes}deg)`
+  }
+  setInterval(analogicClock,1000)
+  setInterval(analogicClockLogo,1000)
```
Comment (JuanequeX) [inline checkClock/app/javascript/packs/analogic.js]: Add a blank line here

**Review by JuanequeX — CHANGES_REQUESTED** [review]

**Comment by JuanequeX** [comment]
Remember that are necessary add a screenshot of your changes, in this case only you showed the route but we don't have any visual references to evaluate the code

File: checkClock/app/assets/stylesheets/home.scss (hunk 1 above)
Comment (dany-herz42) [reply checkClock/app/assets/stylesheets/home.scss]: Fixed

File: checkClock/app/assets/stylesheets/home.scss (hunk 2 above)
Comment (dany-herz42) [reply checkClock/app/assets/stylesheets/home.scss]: Fixed

File: checkClock/app/views/home/index.html.erb (hunk 3 above)
Comment (dany-herz42) [reply checkClock/app/views/home/index.html.erb]: Fixed

File: checkClock/app/javascript/packs/homepage.js (hunk 4 above)
Comment (dany-herz42) [reply checkClock/app/javascript/packs/homepage.js]: Fixed

File: checkClock/app/javascript/packs/analogic.js (hunk 5 above)
Comment (dany-herz42) [reply checkClock/app/javascript/packs/analogic.js]: Fixed

**Review by JuanequeX — APPROVED** [review]
LGTM

File: checkClock/app/helpers/home_helper.rb [hunk 6]
```diff
@@ -0,0 +1,2 @@
+module HomeHelper
+end
```
Comment (YilKanda) [inline checkClock/app/helpers/home_helper.rb:2]: If you don't occupy this helper, you can delete it.

File: checkClock/config/routes.rb [hunk 7]
```diff
@@ -1,3 +1,4 @@
 Rails.application.routes.draw do
+  get 'home/index'
```
Comment (YilKanda) [inline checkClock/config/routes.rb:2]: Are you going to use that route for home? I think you can explain a more concrete route for your dashboard.

```suggestion
Rails.application.routes.draw do
  get 'home/index', as: 'dashboard'
```

**Review by YilKanda — APPROVED** [review]

File: checkClock/app/helpers/home_helper.rb (hunk 6 above)
Comment (dany-herz42) [reply checkClock/app/helpers/home_helper.rb:2]: Ok, we will delete this helper

**Pull request merged**